compiler/backend/collectors.nim

```diff
@@ -0,0 +1,82 @@
+## Implements the "collect" pass. This pass gathers the full AST for each
+## module into a single structure, which is then meant to be consumed by the
+## code-generation orchestrators.
```

Comment:
```suggestion
## code-generation orchestrators (`cbackend`, `jsbackend`, etc).
```

Orchestrator doesn't appear in any identifiers so I figured a bit of clarity here might send folks new to the code in the right direction.